source/extensions/filters/http/cache/simple_http_cache/simple_http_cache.cc

```diff
@@ -153,9 +153,9 @@ SimpleHttpCache::varyLookup(const LookupRequest& request,
       iter->second.body_};
 }
 
-void SimpleHttpCache::varyInsert(
-    const Key& request_key, Http::ResponseHeaderMapPtr&& response_headers, std::string&& body,
-    const std::vector<const Http::HeaderEntry*>& request_vary_headers) {
+void SimpleHttpCache::varyInsert(const Key& request_key,
+                                 Http::ResponseHeaderMapPtr&& response_headers, std::string&& body,
+                                 const Http::RequestHeaderMapPtr& request_vary_headers) {
```

Comment:
const Http::RequestHeaderMap&. There's no reason to take a unique_ptr by const&.